File: Java/MedianofTwoSortedArrays.java
public class MedianofTwoSortedArrays
{
	public double findMedianSortedArrays(int[] nums1,int[] nums2)
	{
		int i=0,j=0,k=(nums1.length+nums2.length)/2;
		int q=(nums1.length+nums2.length)%2;
		int n=0;
		while(true)
		{
			if(nums1[i]>nums2[j])
			{
				if(j<nums2.length)
					j++;
				else
				{
					i++;
					j++;
				}

				if(n==k)
					break;
				else
					n++;
			}
			else
			{
				if(i<nums1.length)
					i++;
				else
				{
					j++;
					i++;
				}
				if(n==k)
					break;
				else
					n++;
			}
		}
		if(q==1)
		{
			if(i>=nums1.length)
				return (double)nums2[j];
			else if(j>=nums2.length)
				return (double)nums1[i];
			else
				return nums1[i]>nums2[j]?(double)nums1[i]:(double)nums2[j];

		}
		if(q==0)
		{
			if(i>=nums1.length)
			{
				return (double)(nums2[j]+nums2[j-1])/2;
			}
			else if(j>=nums2.length)
			{
				return (double)(nums1[i]+nums2[i-1])/2;
			}
			else
			{
				return (nums1[i]+nums2[j])/2;
			}
		}
                return 0;
	}
}

File: C/MedianofTwoSortedArrays.c

//merge
//count the number of smaller half part

double findMedianSortedArrays(int *nums1, int nums1Size, int *nums2, int nums2Size)
{
	int sum = nums1Size + nums2Size, count = sum / 2, midleft, midright;
	for (int i = 0, j = 0; count >= 0 && (i < nums1Size || j < nums2Size);)
	{
		if (i == nums1Size)
			midright = nums2[j++];
		else if (j == nums2Size)
			midright = nums1[i++];
		else
			midright = (nums1[i] < nums2[j] ? nums1[i++] : nums2[j++]);
		if (--count == 0)
			midleft = midright;
	}
	return sum % 2 == 0 ? (midleft + midright) / 2.0 : midright;
}